tensorflow/python/keras/backend_test.py

```diff
@@ -234,10 +235,13 @@ def test_function_tf_fetches(self):
       x_placeholder = keras.backend.placeholder(shape=())
       y_placeholder = keras.backend.placeholder(shape=())
 
+      run_options = config_pb2.RunOptions(
+        trace_level=config_pb2.RunOptions.NO_TRACE)
       f = keras.backend.function(inputs=[x_placeholder, y_placeholder],
                                  outputs=[x_placeholder + y_placeholder],
                                  updates=[(x, x_placeholder + 1.)],
-                                 fetches=[keras.backend.update(y, 5.)])
+                                 fetches=[keras.backend.update(y, 5.)],
+                                 options=run_options)
```

Comment:
I'd argue that this test is inadequate because it doesn't verify the effect of sending the run_options.
I suggest you use both the `options` and `run_metadata` keywords here and set `output_partition_graphs` of `RunOptions` to `True`. Then you can verify that the `partition_graphs` of `run_metadata` is set after the call.

Reply:
I agree that it's better to add `run_metadata` keyword, however I'm afraid `_make_callable_from_options` doesn't support `run_metadata` yet.

```python
callable_fn = session._make_callable_from_options(callable_opts)
```
https://github.com/facaiy/tensorflow/blob/a5840964e0eb422fbe73dd3738c8d14c1147276f/tensorflow/python/keras/backend.py#L2853

The one solution I can see is to replace `session._make_callable_from_options` by `session.make_callable`, how do you think it?  cc @fchollet @mrry

Reply:
@facaiy run_metadata support for TF_SessionMakeCallable has been added by @mrry https://github.com/tensorflow/tensorflow/commit/574a85178942418f5531c215b74729b38b4499d2

Reply:
Thank you for letting me know. I'd like to add the argument in the weekend.